tests/persistSync.test.tsx

```diff
@@ -28,118 +27,96 @@ describe('persist middleware with sync configuration', () => {
         {
           name: 'test-storage',
           getStorage: () => storage,
-          onRehydrateStorage: () => (state, error) => {
-            postRehydrationCallbackCallCount++
-            expect(error).toBeUndefined()
-            expect(state?.count).toBe(42)
-            expect(state?.name).toBe('test-storage')
-          },
+          onRehydrateStorage: () => spy,
         }
       )
     )
 
     expect(useStore.getState()).toEqual({ count: 42, name: 'test-storage' })
-    expect(postRehydrationCallbackCallCount).toBe(1)
+    expect(spy).toBeCalledWith({ count: 42, name: 'test-storage' }, undefined)
   })
 
   it('can throw rehydrate error', () => {
-    let postRehydrationCallbackCallCount = 0
-
     const storage = {
       getItem: () => {
         throw new Error('getItem error')
       },
       setItem: () => {},
     }
 
+    const spy = jest.fn()
     create(
       persist(() => ({ count: 0 }), {
         name: 'test-storage',
         getStorage: () => storage,
-        onRehydrateStorage: () => (_, e) => {
-          postRehydrationCallbackCallCount++
-          expect(e?.message).toBe('getItem error')
-        },
+        onRehydrateStorage: () => spy,
       })
     )
 
-    expect(postRehydrationCallbackCallCount).toBe(1)
+    expect(spy).toBeCalledWith(undefined, new Error('getItem error'))
   })
 
   it('can persist state', () => {
-    let setItemCallCount = 0
+    const setItemSpy = jest.fn()
+    const onRehydrateStorageSpy = jest.fn()
 
     const storage = {
       getItem: () => null,
-      setItem: (name: string, value: string) => {
-        setItemCallCount++
-        expect(name).toBe('test-storage')
-        expect(value).toBe(
-          JSON.stringify({
-            state: { count: 42 },
-            version: 0,
-          })
-        )
-      },
+      setItem: setItemSpy,
     }
 
     const useStore = create(
       persist(() => ({ count: 0 }), {
         name: 'test-storage',
         getStorage: () => storage,
-        onRehydrateStorage: () => (_, error) => {
-          expect(error).toBeUndefined()
-        },
+        onRehydrateStorage: () => onRehydrateStorageSpy,
       })
     )
 
     expect(useStore.getState()).toEqual({ count: 0 })
     useStore.setState({ count: 42 })
     expect(useStore.getState()).toEqual({ count: 42 })
-    expect(setItemCallCount).toBe(1)
+    expect(setItemSpy).toBeCalledWith(
+      'test-storage',
+      JSON.stringify({ state: { count: 42 }, version: 0 })
+    )
+    expect(onRehydrateStorageSpy).toBeCalledWith(undefined, undefined)
   })
 
   it('can migrate persisted state', () => {
-    let migrateCallCount = 0
-    let setItemCallCount = 0
+    const setItemSpy = jest.fn()
+    const onRehydrateStorageSpy = jest.fn()
+    const migrateSpy = jest.fn(() => ({ count: 99 }))
```

Comment:
Note that in this case the spy is not simply to verify that it has been called, it needs to return a value and is actually is a crucial part of the test itself. 

Which, one could argue, makes the test less expressive. Do you still want to use `jest.fn()`, here, or should those cases be left as they are?

Reply:
I think the `jest.fn` is fine. I've always used mocked functions to return static values, so I don't really find it disturbing